R/checks.R

```diff
@@ -254,6 +254,40 @@ check_req_labelled_metadata_fields <- function(supertbl, call = caller_env()) {
 }
 
 
+#' @title
+#' Check that parsed labels are not duplicated
+#'
+#' @importFrom cli cli_warn qty
+#' @importFrom rlang caller_env
+#'
+#' @param parsed_labs a vector of parsed labels produced by `parse_labels()`
+#' @param field_name the name of the field associated with the labels to use in the warning message
+#' @param call the calling environment to use in the error message. The parent of calling environment
+#' by default because this check usually occurs 2 frames below the relevant context for the user
+#'
+#' @return
+#' a warning message alerting specifying the duplicate labels and REDCap field affected
+#'
+#' @keywords internal
+check_parsed_labels <- function(parsed_labs, field_name, call = caller_env(n = 2)) {
+  if (any(duplicated(parsed_labs))) {
+    dups <- parsed_labs[duplicated(parsed_labs)]
+
+    msg <- c(
+      "!" = "Multiple values are mapped to the {qty(dups)} label{?s} {.code {dups}} in field {.code {field_name}}",
+      "i" = "Consider making the labels for {.code {field_name}} unique in your REDCap project"
```

Comment:
Probably not but I'm unsure of the best way to fix that. I'd like to avoid a solution that requires the check to know whether or not we stripped html tags because this happens inside `parse_labels`. A simple alternative is to just always include a note saying "`REDCapTidieR` removes html tags from labels which may have caused the duplication."

What do you think?

Reply:
It depends how worthwhile we find alerting users to HTML. I'm just thinking in Roni's case the current error would have probably been confusing anyway since he/someone might have thought his labels were unique, but it was actually on our end that we made them duplicates.

It would be more work, and may not be worthwhile, but possibly we could expand `strip_html_field_embedding` to also `detect_html_field_embedding` and if triggered (or is `TRUE`, or whatever), append the `cli` message to include the note you suggested.

Reply:
Discussed checking for empty values. If duplicate values are due to only empty values then only show warning related to empty labels and not also the duplicate label warning.

Add the following if this is because we removed HTML?
`"i" = "This may happen if the label only contains HTML"`